content/en/blog/2023/end-user-q-and-a-04.md

```diff
@@ -0,0 +1,441 @@
+use them in conjunction with each other," he says, "as long as you have 
+propagators set up correctly."
+
+After setting up the propagators correctly, they made sure all their plugins 
+(which are now open source) worked. They had to revert a few times from their 
+staging environment, but didn't encounter any major problems aside from a bug 
+that he missed. 
+
+"I had to implement a custom sampler, which is ten times easier with OTel than 
+it was with OpenTracing," he says. "I was able to get rid of a thousand lines of 
+code and some dangerous hacks, so that was a really good thing." 
+
+### How to start a migration
+
+"I started with really small services my team owned with really low traffic, but 
+enough for it to be constant," Jacob says. "The reason you want to pick a service 
+like this is that if it's too low traffic, like one request every 10 minutes, you 
+have to worry about sample rates, [and] you may not have a lot of data to compare 
+against – that’s the big thing: you need to have some data to compare against."
+
+He had written a script early on for their metrics migration that queried 
+different build tags that were on all their metrics. If the standard deviation 
+for the newer build tag is greater than 1 compared to the previous release, that 
+could signal an issue with your instrumentation library. 
+
+"Another thing I had to check was that all the attributes were still present 
+before and after migration, which is another thing that matters," Jacob notes. 
+Sometimes they weren't, as in the case of Statsd automatically adding something 
+they didn't care about; those could be safely ignored.
+
+For tracing, Jacob says, "I picked a service that had both internal-only traces 
+(stayed within a single service) and traces that spanned multiple services with 
+different types of instrumentation, so from Envoy to OTel to OpenTracing." 
+
+He explains, "What you want to see is that the trace before has the same 
+structure as the trace after. So I made another script that checked that those 
+structures were relatively the same and that they all had the same attributes as 
+well... That’s the point of the tracing migration – what matters is that all the 
+attributes stayed the same."
+
+### When data goes missing
+
+"The ‘why it’s missing’ stories are the really complicated ones," says Jacob. 
+Sometimes, it's as simple as forgetting "to add something somewhere," but other 
+times, there could be an upstream library that doesn't emit what you expected 
+for OTel. 
+
+He tells a story about the time he  migrated their grpc util package (which is 
+now in Go contrib) and found an issue with propagation. 
+
+"I was trying to understand what’s going wrong here. When I looked at the code – 
+this tells you how early I was doing this migration – where there was supposed to 
+be a propagator, there was just a 'TODO' from Alex Boten," he shares. "It just 
```

Comment:
nit: can we remove alex's name here? no need for the callout 😅

Reply:
😂 Yes, absolutely, I thought it was funny @codeboten